File: 530-minimum-absolute-difference-in-bst/minimum-absolute-difference-in-bst.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public int GetMinimumDifference(TreeNode root) {
        List<int> values = new List<int>();
        InOrderTraversal(root, values);
        
        int minDifference = int.MaxValue;
        
        for (int i = 1; i < values.Count; i++) {
            int difference = values[i] - values[i - 1];
            if (difference < minDifference) {
                minDifference = difference;
            }
        }
        
        return minDifference;
    }
    
    private void InOrderTraversal(TreeNode node, List<int> values) {
        if (node == null) return;
        
        InOrderTraversal(node.left, values);
        values.Add(node.val);
        InOrderTraversal(node.right, values);
    }
}

File: 530-minimum-absolute-difference-in-bst/minimum-absolute-difference-in-bst.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var getMinimumDifference=function(root){
    let prev=null, minDiff=Infinity;
    const inOrderTraversal=(node)=>{
        if(!node)return;
        inOrderTraversal(node.left);
        if(prev !==null){minDiff=Math.min(minDiff,node.val-prev);}
        prev=node.val;
        inOrderTraversal(node.right);
    };
    inOrderTraversal(root);
    return minDiff;
};